Source/WTF/Configurations/modulemap.toml

```diff
@@ -0,0 +1,43 @@
+module-name = 'wtf'
+config-macros = ['UCHAR_TYPE']
+requirements = ['cplusplus11']
+
+# Swift's C++ import struggles with some of the templates found in
+# WTF. To avoid Swift trying to import everything, we divide WTF
+# into lots of submodules. This is why we can't use features such
+# as an umbrella directory.
+one-submodule-per-header = true
+
+# This header depends on surrounding preprocessor context.
+textual-headers = ['spi/mac/MetadataSPI.h']
+
+# Some WTF modules can be used in C contexts
+[module.Assertions]
+attributes = ['extern_c']
+requirements = []
+[module.AutodrainedPool]
+requirements = ['cplusplus11', '!objc']
+[module.Compiler]
+attributes = ['extern_c']
+requirements = []
+[module.ExportMacros]
+attributes = ['extern_c']
+requirements = []
+[module.IOSurfaceSPI]
+attributes = ['extern_c']
+requirements = []
+[module.HashMap]
+# These two headers are interdependent so need to be colocated
+# in one submodule.
```

Comment:
What does it mean? There are many header dependencies, for example, Vector is relying on FastMalloc.h. But they are not having one module for them. So what is the difference?

Reply:
I'll elaborate on this when I come back to this PR. The key part is _inter_dependent - that is, they depend on each other in a circular fashion.